Template/update320

anoronh4 opened the pull request:
Updating the template to nf-core/tools version 3.2.0. i was able to do this with `nf-core pipelines sync`, which means we are probably finally free from doing manual synchronization.

also fixing `nextflow_schema.json` and any nf-core lint errors

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/mskcc/forte/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the mskcc/forte _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

anoronh4 committed:
fix linting errors

anoronh4 committed:
Template update for nf-core/tools version 3.2.0

anoronh4 committed:
Merge branch 'TEMPLATE' into template/update320

anoronh4 committed:
update changelog

anoronh4 committed:
remove redundant process.shell from tests config

anoronh4 committed:
fix nf-core linting errors

anoronh4 committed:
change repo address in modules.json to https protocol

anoronh4 committed:
fix awsfulltest workflow for nf-core tools compliance

anoronh4 committed:
fix linting error with prettier

anoronh4 committed:
Merge branch 'bugfix/nextflow_schema' of github.com:mskcc/forte into bugfix/nextflow_schema

anoronh4 committed:
change ensembl_version type to integer in schema

anoronh4 committed:
remove clinicalgenes parameter from closure

johnoooh commented on README.md:
```diff
@@ -87,7 +87,9 @@ If you would like to contribute to this pipeline, please see the [contributing g
-<!-- If you use mskcc/forte for your analysis, please cite it using the following doi: [10.5281/zenodo.XXXXXX](https://doi.org/10.5281/zenodo.XXXXXX) --><!-- TODO nf-core: Add bibliography of tools and data used in your pipeline -->
+<!-- If you use mskcc/forte for your analysis, please cite it using the following doi: [10.5281/zenodo.XXXXXX](https://doi.org/10.5281/zenodo.XXXXXX) -->
+
+<!-- TODO nf-core: Add bibliography of tools and data used in your pipeline -->
```
You did this already, can get rid of this TODO

pintoa1-mskcc reviewed (changes requested):
This PR has the same bug as the previous [template/update branch ](https://github.com/mskcc/forte/pull/133#issuecomment-2695287482), should be a simple fix and I will test again 

The solution was: 
> > after I updated nextflow to work with this branch, something is wrong with the baits file now?
> > error I'm getting
> > > * --baits ({"idt_v2":{"baits":"/juno/work/ccs/cmopipeline/forte/GRCh38_probes/xgen-exome-hyb-panel-v2-probes-hg38.bed","targets":"/juno/work/ccs/cmopipeline/forte/GRCh38_probes/xgen-exome-hyb-panel-v2-targets-hg38.bed"}}): Value is [object] but should be [string]
> > 
> > 
> > Curious if its something to do with the change ifrom having params in getGenomeAttribute to removing it in this:
> > https://github.com/mskcc/forte/blob/db0c18fb533482c6a42bacb9f75350477f6d306d/main.nf#L22
> 
> this error is caused by incorrect configuration in `nextflow_schema.json`. nf-schema was looking through the configs and expecting params.baits to be a `"string"` and not an `"object"`. we definitely want the latter so i've updated that, seems to be working.

anoronh4 committed:
remove completed TODO

anoronh4 committed:
updated baits schema definition in nextflow_schema.json

anoronh4 commented:
> This PR has the same bug as the previous [template/update branch ](https://github.com/mskcc/forte/pull/133#issuecomment-2695287482), should be a simple fix and I will test again
> 
> The solution was:
> 
> > > after I updated nextflow to work with this branch, something is wrong with the baits file now?
> > > error I'm getting
> > > > * --baits ({"idt_v2":{"baits":"/juno/work/ccs/cmopipeline/forte/GRCh38_probes/xgen-exome-hyb-panel-v2-probes-hg38.bed","targets":"/juno/work/ccs/cmopipeline/forte/GRCh38_probes/xgen-exome-hyb-panel-v2-targets-hg38.bed"}}): Value is [object] but should be [string]
> > > 
> > > 
> > > Curious if its something to do with the change ifrom having params in getGenomeAttribute to removing it in this:
> > > https://github.com/mskcc/forte/blob/db0c18fb533482c6a42bacb9f75350477f6d306d/main.nf#L22
> > 
> > 
> > this error is caused by incorrect configuration in `nextflow_schema.json`. nf-schema was looking through the configs and expecting params.baits to be a `"string"` and not an `"object"`. we definitely want the latter so i've updated that, seems to be working.

good catch, for this PR i regenerated to the whole nextflow_schema.json using the nf-core tool-kit, which overwrote how I had set up this param in the last PR. the nf-core lint tool never catches this for some reason!
